### PR TITLE
Add nonce

### PR DESCRIPTION
**Pull request by zachmargolis**
nonce is now required

### CODE REVIEW

**Comment by jessieay** [comment]
is there a test we can write to confirm this works? not sure what is requiring this, exactly.

**Comment by zachmargolis** [comment]
😬 we don't have a test for this yet.

The thing that's requiring this is the IDP, as part of https://github.com/18F/identity-idp/pull/1236

**Comment by zachmargolis** [comment]
ok @jessieay updated with specs!

File: .reek [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  TooManyInstanceVariables:
+    enabled: false
+  TooManyMethods:
+    enabled: false
+  TooManyStatements:
+    enabled: false
+  UtilityFunction:
+    enabled: false
+exclude_paths:
+  - db/migrate
```
Comment (jessieay) [inline .reek]: glad we are adding rubocop and reek!!! Can we do this in a separate PR, though? Feels unrelated to this exact change

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+language: ruby
+matrix:
+  fast_finish: true
+notifications:
+  email: false
+rvm:
+  - ruby-2.3.3
+sudo: false
+script:
+  - make test
```
Comment (jessieay) [inline .travis.yml]: add travis config separately, too?

**Comment by zachmargolis** [comment]
Ok, I will break this back into separate PRs :)
